elastic/kibana
# CHIPS cookie support

## Summary

Added CHIPS support. `statehood` doesn't support partitioned cookie out of the box. We can leverage [contextualize](https://github.com/hapijs/statehood/blob/master/lib/index.js#L35) function in statehood with a trick, modifying `isSameSite` property. `Partitioned` attribute is appended only if embedding is not disabled.

## How to test
1. Add to kibana config:
```yml
server.securityResponseHeaders.disableEmbedding: false
xpack.security.sameSiteCookies: 'None'
xpack.security.secureCookies: true
```
2. ES and Kibana need to run over https to test that, because of `SameSite=None` settings.
Check the `sid` cookie was set with `Partitioned` attribute.

`Set-Cookie` header has `Partitioned` attribute.

![Screenshot 2024-07-10 at 17 53 24](https://github.com/user-attachments/assets/5ddbe7c5-8648-4552-8697-504a32a42bda)

Stored cookie has a `Partition Key`

<img width="1233" alt="Screenshot 2024-07-10 at 18 04 13" src="https://github.com/user-attachments/assets/cc0c453e-e33f-4999-bcff-d2acd3a0fcd2">


### Checklist


__Fixes: https://github.com/elastic/kibana/issues/180974__

### Release Notes
Added CHIPS cookie support.
